clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -727,12 +733,22 @@ private void validateOffsetsAsync(Map<TopicPartition, SubscriptionState.FetchPos
                 return;
             }
 
-            subscriptions.setNextAllowedRetry(dataMap.keySet(), time.milliseconds() + requestTimeoutMs);
-
             final Map<TopicPartition, Metadata.LeaderAndEpoch> cachedLeaderAndEpochs = partitionsToValidate.entrySet()
                     .stream()
                     .collect(Collectors.toMap(Map.Entry::getKey, entry -> entry.getValue().currentLeader));
 
+            NodeApiVersions nodeApiVersions = apiVersions.get(node.idString());
+            if (nodeApiVersions == null || nodeApiVersions.latestUsableVersion(ApiKeys.OFFSET_FOR_LEADER_EPOCH) < 3) {
```

Comment:
Maybe we should have a static method in `OffsetForLeaderEpochRequest` that this class can pass the version to which returns `true` if it's 3 and higher.